posthog/models/historical_version.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+import json
+
+from django.db import models
+from django.utils import timezone
+
+from posthog.models import FeatureFlag
+from posthog.models.utils import UUIDModel
+
+
+class HistoricalVersionJSONEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, models.base.ModelState):
+            return None
+        if isinstance(obj, FeatureFlag):
+            return obj.__dict__
+        if isinstance(obj, datetime.datetime):
+            return obj.isoformat()
+
+        return json.JSONEncoder.default(self, obj)
+
+
+class HistoricalVersion(UUIDModel):
```

Comment:
Any estimate on the size of this table? I assume there are actually very few feature flags, and not that many of the other objects that we will be included?

Reply:
The ones we've thought about are here https://github.com/PostHog/posthog/issues/8545#issuecomment-1035429062

Models that do have a risk of large numbers are event definitions and property definitions (IMO). I'd be in favour of either not tracking their history, or leaving them to much later so we have real data to consider for performance impact